orc/src/test/java/org/apache/iceberg/orc/TestORCSchemaUtil.java

```diff
@@ -209,4 +210,39 @@ public void testInvalidTypePromotions() {
           ORCSchemaUtil.buildOrcProjection(evolveSchema, orcSchema);
         });
   }
+
+  @Test
+  public void testSkipNonIcebergColumns() {
+    TypeDescription schema = TypeDescription.createStruct();
+    TypeDescription intCol = TypeDescription.createInt();
+    intCol.setAttribute(ICEBERG_ID_ATTRIBUTE, "1");
+    TypeDescription listCol = TypeDescription
+        .createList(TypeDescription.createMap(TypeDescription.createString(), TypeDescription.createDate()));
+    listCol.setAttribute(ICEBERG_ID_ATTRIBUTE, "2");
+    schema.addField("intCol", intCol);
+    schema.addField("listCol", listCol);
+    TypeDescription stringKey = TypeDescription.createString();
+    stringKey.setAttribute(ICEBERG_ID_ATTRIBUTE, "3");
+    TypeDescription booleanVal = TypeDescription.createBoolean();
+    booleanVal.setAttribute(ICEBERG_ID_ATTRIBUTE, "4");
+    TypeDescription mapCol = TypeDescription.createMap(stringKey, booleanVal);
+    mapCol.setAttribute(ICEBERG_ID_ATTRIBUTE, "5");
+    schema.addField("mapCol", mapCol);
+
+    Schema icebergSchema = ORCSchemaUtil.convert(schema);
+    assertEquals(2, icebergSchema.asStruct().fields().size());
```

Comment:
I think this should create an Iceberg schema and assert the two are equal (structs implement equals, so you have to check `assertEquals(expected.asStruct(), converted.asStruct())`.

Reply:
Done. I've changed the test to compare structs.